infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -141,280 +158,315 @@ private boolean isBootstrapped(CorfuMsg msg, ChannelHandlerContext ctx, IServerR
         return true;
     }
 
+    private boolean isBootstrapped(HeaderMsg reqHeader, ChannelHandlerContext ctx, IServerRouter r) {
```

Comment:
I would pass the `RequestMsg` and extract the header just before it is needed. Passing header around seems a little strange.